Leanplum/Cache/LPCache.m

```diff
@@ -0,0 +1,105 @@
+//
+//  LPCache.m
+//  Leanplum
+//
+//  Created by Grace on 7/23/19.
+//  Copyright © 2019 Leanplum. All rights reserved.
+//
+
+#import "LPCache.h"
+#import "LPAES.h"
+
+@implementation LPCache
+
++ (instancetype)sharedCache {
+    static LPCache *sharedCache = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedCache = [[self alloc] init];
+    });
+    return sharedCache;
+}
+
+- (NSData *)encrypt:(NSData *)data {
+    return [LPAES encryptedDataFromData:data];
+}
+
+- (NSData *)decrypt:(NSData *)data {
+    return [LPAES decryptedDataFromData:data];
+}
+
+- (NSData *)encodeIntoBinary:(NSArray *)data {
+    return [NSData new];
+}
+
+- (NSArray *)decodeFromBinary:(NSData *)data {
+    return [NSArray new];
+}
+
+- (void)persist:(NSData *)data forKey:(NSString *)key {
+    
```

Comment:
adding `//TODO:` here and in other places that are incomplete, will help this PR be more self contained.